addons/sale_timesheet/models/sale_order.py

```diff
@@ -85,22 +102,38 @@ def _get_prepaid_service_lines_to_upsell(self):
 
     def action_view_timesheet(self):
         self.ensure_one()
+        if not self.order_line:
+            return {'type': 'ir.actions.act_window_close'}
+
         action = self.env["ir.actions.actions"]._for_xml_id("sale_timesheet.timesheet_action_from_sales_order")
-        action['context'] = {
-            'search_default_billable_timesheet': True
+        default_sale_line = next(sale_line for sale_line in self.order_line if sale_line.is_service and sale_line.product_id.service_policy in ['ordered_prepaid', 'delivered_timesheet'])
+        context = {
+            'search_default_billable_timesheet': True,
+            'default_is_so_line_edited': True,
+            'default_so_line': default_sale_line.id,
         }  # erase default filters
-        if self.order_line:
-            tasks = self.order_line.task_id._filter_access_rules_python('write')
-            if tasks:
-                action['context']['default_task_id'] = tasks[0].id
-            else:
-                projects = self.order_line.project_id._filter_access_rules_python('write')
-                if projects:
-                    action['context']['default_project_id'] = projects[0].id
-        if self.timesheet_count > 0:
-            action['domain'] = [('so_line', 'in', self.order_line.ids)]
+
+        tasks = self.order_line.task_id._filter_access_rules_python('write')
+        if tasks:
+            context['default_task_id'] = tasks[0].id
         else:
-            action = {'type': 'ir.actions.act_window_close'}
+            projects = self.order_line.project_id._filter_access_rules_python('write')
+            if projects:
+                context['default_project_id'] = projects[0].id
+            elif self.project_ids:
+                context['default_project_id'] = self.project_ids[0].id
+        action.update({
+            'context': context,
+            'domain': [('so_line', 'in', self.order_line.ids)],
+            'help': [_("""
+                <p class="o_view_nocontent_smiling_face">
+                    No activities found. Let's start a new one!
+                </p><p>
+                    Track your working hours by projects every day and invoice this time to your customers.
+                </p>
+            """)]
```

Comment:
The array is useless indeed, but i think we can not use the markup (if markup is what i think it is : the _t ) because we're sending an action back. At least i did not find any example in the code with a _t.